### PR TITLE
fix(ingestion/transformer): tranformer to replace the externalUrl in container properties

### PR DESCRIPTION
**Pull request by sagar-salvi-apptware**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Summary by CodeRabbit

- **New Features**
  - Introduced enhanced documentation for the "Replace ExternalUrl" functionality, with clearer distinctions between dataset and container transformations.
  - Added a new transformer type for handling external URLs within container properties.
  - Implemented new classes and methods to improve handling of external URLs in dataset and container contexts.

- **Bug Fixes**
  - Updated existing tests to reflect changes in transformer functionality, ensuring accuracy and reliability.

- **Documentation**
  - Enhanced documentation with detailed configuration options and examples for new transformers.

### CODE REVIEW

**Review by eboneil — COMMENTED** [review]
General feedback: instead of making this a new transformer, or making this a configuration of the transformer, can we just automatically detect it within the existing transformer? So like we are doing with datasets today, if there are containers, just go and make the replacement?

**Comment by sagar-salvi-apptware** [comment]
> General feedback: instead of making this a new transformer, or making this a configuration of the transformer, can we just automatically detect it within the existing transformer? So like we are doing with datasets today, if there are containers, just go and make the replacement?

Problem with the current structure of transformer we cant have 2 different aspect, we can have multiple entity types but of same aspects.

**Review by eboneil — APPROVED** [review]
Thank you Sagar!

File: metadata-ingestion/docs/transformer/dataset_transformer.md [hunk 1]
```diff
@@ -953,7 +953,7 @@ Then define your class to return a list of custom properties, for example:
           add_properties_resolver_class: "<your_module>.<your_class>"
     ```
 
-## Replace ExternalUrl
+## Replace ExternalUrl Dataset
```
[the remark this answers is not in the document]
Comment (sagar-salvi-apptware) [reply metadata-ingestion/docs/transformer/dataset_transformer.md:956]: No need

File: metadata-ingestion/src/datahub/ingestion/transformer/replace_external_url.py [hunk 2]
```diff
@@ -6,18 +6,28 @@
...
     input_pattern: str
     replacement: str
 
 
-class ReplaceExternalUrl(DatasetPropertiesTransformer):
-    """Transformer that clean the ownership URN."""
+class ReplaceUrl:
+    def replace_url(self, pattern: str, replacement: str, external_url: str) -> str:
+        pattern_obj = re.compile(pattern)
+        return re.sub(pattern_obj, replacement, external_url)
```
[the remark this answers is not in the document]
Comment (sagar-salvi-apptware) [reply metadata-ingestion/src/datahub/ingestion/transformer/replace_external_url.py:26]: Not needed

File: metadata-ingestion/docs/transformer/dataset_transformer.md [hunk 3]
```diff
@@ -971,6 +971,24 @@
       replacement: "sub"
 ```
 
+## Replace ExternalUrl Container
```
[the remark this answers is not in the document]
Comment (sagar-salvi-apptware) [reply metadata-ingestion/docs/transformer/dataset_transformer.md:974]: Not needed

**Pull request merged**